tests/test_main.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+import pytest
+import pytest_mock
+
+import pyproject_api.__main__
+from pyproject_api._frontend import EditableResult, SdistResult, WheelResult
+from pyproject_api._via_fresh_subprocess import SubprocessFrontend
+
+
+@pytest.mark.parametrize(
+    ("cli_args", "srcdir", "hooks"),
+    [
+        (
+            [],
+            Path.cwd(),
+            ["build_sdist", "build_wheel"],
+        ),
+        (
+            ["foo"],
+            Path("foo"),
+            ["build_sdist", "build_wheel"],
+        ),
+        (
+            ["-s"],
+            Path.cwd(),
+            ["build_sdist"],
+        ),
+        (
+            ["-w"],
+            Path.cwd(),
+            ["build_wheel"],
+        ),
+        (
+            ["-e"],
+            Path.cwd(),
+            ["build_editable"],
+        ),
+        (
+            ["-s", "-w"],
+            Path.cwd(),
+            ["build_sdist", "build_wheel"],
+        ),
+    ],
+)
+def test_parse_args(mocker: pytest_mock.MockerFixture, cli_args: list[str], srcdir: Path, hooks: list[str]) -> None:
+    out = srcdir / "dist"
+
+    subprocess_frontend = mocker.MagicMock(spec=SubprocessFrontend)
+    subprocess_frontend.create_args_from_folder.return_value = (srcdir, (), "foo.bar", "baz", (), True)
+    subprocess_frontend.build_sdist.return_value = SdistResult(sdist=out / "foo.whl", out="", err="")
+    subprocess_frontend.build_wheel.return_value = WheelResult(wheel=out / "foo.whl", out="", err="")
+    subprocess_frontend.build_editable.return_value = EditableResult(wheel=out / "foo.whl", out="", err="")
+    mocker.patch("pyproject_api.__main__.SubprocessFrontend", return_value=subprocess_frontend)
+
+    pyproject_api.__main__.main(cli_args)
+
+    # FIXME: Something weird is going on here and I can't figure out what. I suspect it's something to do with us
+    # mocking the whole class and this being a classmethod
+    # subprocess_frontend.create_args_from_folder.assert_called_once_with(srcdir)
```

Comment:
We should fix this before merging

Reply:
Figured it out. Class != object. Duh :see_no_evil: